source/framework/tools/src/TRestTools.cxx

```diff
@@ -1070,7 +1068,11 @@ int TRestTools::DownloadRemoteFile(string remoteFile, string localFile) {
     if ((string)url.GetProtocol() == "https" || (string)url.GetProtocol() == "http") {
         string path = TRestTools::SeparatePathAndName(localFiletmp).first;
         if (!TRestTools::fileExists(path)) {
-            system(("mkdir -p " + path).c_str());
+            int r = system(("mkdir -p " + path).c_str());
+            if (r != 0) {
+                std::cout << "mkdir failed to create directory: " << path << std::endl;
+                return -1;
+            }
```

Comment:
Since we are using C++17 we should use the `<filesystem>` API whenever we need to do interface with the filesystem such as in this case. (may need to `#include <filesystem>`).

```suggestion
            if (!filesystem::create_directories(path)){
                std::cerr << "mkdir failed to create directory: " << path << std::endl;
                return -1;
            }
```

Reply:
Also I think we should use `std::cerr` to use the standard error stream instead of the standard output whenever there is an error.